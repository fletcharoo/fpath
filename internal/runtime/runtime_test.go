package runtime_test

import (
	"os"
	"testing"

	"github.com/fletcharoo/fpath/internal/lexer"
	"github.com/fletcharoo/fpath/internal/parser"
	"github.com/fletcharoo/fpath/internal/runtime"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestMain(m *testing.M) {
	r := m.Run()
	snaps.Clean(m, snaps.CleanOpts{Sort: true})
	os.Exit(r)
}

func Test_Eval(t *testing.T) {
	testCases := map[string]struct {
		query string
		input any
	}{
		"number": {
			query: "2",
		},
	}

	for name, tc := range testCases {
		t.Run(name, func(t *testing.T) {
			lex := lexer.New(tc.query)
			expr, err := parser.New(lex).Parse()
			if err != nil {
				t.Fatalf("Unexpected parser error: %s", err)
			}

			result, err := runtime.Eval(expr, err)
			if err != nil {
				t.Fatalf("Unexpected runtime error: %s", err)
			}

			resultDecoded, err := result.Decode()
			if err != nil {
				t.Fatalf("Failed to decode result: %s", err)
			}

			snaps.MatchSnapshot(t, resultDecoded)
		})
	}
}
