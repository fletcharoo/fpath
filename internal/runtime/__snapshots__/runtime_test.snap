
[Test_Eval/add - 1]
float64(5)
---

[Test_Eval/multiply - 1]
float64(6)
---

[Test_Eval/number - 1]
float64(2)
---
